src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/WhyCommand/WhyCommandRunner.cs

```diff
@@ -0,0 +1,548 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Build.Evaluation;
+using NuGet.ProjectModel;
+using NuGet.Shared;
+
+namespace NuGet.CommandLine.XPlat
+{
+    internal class WhyCommandRunner : IWhyCommandRunner
+    {
+        private const string ProjectName = "MSBuildProjectName";
+        private const string ProjectAssetsFile = "ProjectAssetsFile";
+
+        // Dependency graph console output symbols
+        private const string ChildNodeSymbol = "├─── ";
+        private const string LastChildNodeSymbol = "└─── ";
+
+        private const string ChildPrefixSymbol = "│    ";
+        private const string LastChildPrefixSymbol = "     ";
+
+        private const string BoldTextEscapeChars = "\u001b[1m";
+        private const string ResetTextEscapeChars = "\u001b[0m";
+
+        /// <summary>
+        /// Executes the 'why' command.
+        /// </summary>
+        /// <param name="whyCommandArgs">CLI arguments for the 'why' command.</param>
+        /// <returns></returns>
+        public Task ExecuteCommand(WhyCommandArgs whyCommandArgs)
+        {
+            var msBuild = new MSBuildAPIUtility(whyCommandArgs.Logger);
+
+            string targetPackage = whyCommandArgs.Package;
+
+            IEnumerable<string> projectPaths = Path.GetExtension(whyCommandArgs.Path).Equals(".sln")
+                                                    ? MSBuildAPIUtility.GetProjectsFromSolution(whyCommandArgs.Path).Where(f => File.Exists(f))
+                                                    : new List<string>() { whyCommandArgs.Path };
+
+            foreach (var projectPath in projectPaths)
+            {
+                Project project = MSBuildAPIUtility.GetProject(projectPath);
```

Comment:
Maybe we can enhance the MSBuildAPIUtility to just return the ProjectAndSolution type instead of selecting only the paths in `GetProjectsFromSolution`. That way, you wouldn't need convert the paths back to projects, here. 

I'm not familiar with these types, so let's see if @jeffkl has opinions on this.